src/sql/workbench/services/executionPlan/common/executionPlanService.ts

```diff
@@ -27,19 +27,46 @@ export class ExecutionPlanService implements IExecutionPlanService {
 		this._providerRegisterEvent = this._onProviderRegister.event;
 	}
 
-	private async ensureCapabilitiesRegistered(): Promise<void> {
-		let providers = Object.keys(this._capabilitiesService.providers);
-		if (!providers) {
-			await new Promise<void>(resolve => {
-				this._capabilitiesService.onCapabilitiesRegistered(e => {
-					resolve();
+	/**
+	 * This ensures that the capabilities service has registered the providers to handle execution plan requests.
+	 */
+	private async ensureCapabilitiesRegistered(providerId?: string): Promise<void> {
```

Comment:
This shouldn't be optional - you're always passing in a providerId (or if you aren't then you need to fix up the parameter in getExecutionPlanProvider)

```suggestion
	private async ensureCapabilitiesRegistered(providerId: string): Promise<void> {
```